scripts/test.py

```diff
@@ -80,27 +80,27 @@ def test_fuzzy(pattern_length: int, haystack_length: int, variability: int):
 def test_slices():
     native = get_random_string(length=10)
     big = Str(native)
-    assert native[0:5] == str(big.sub(0, 5))
-    assert native[5:10] == str(big.sub(5, 10))
-
-    assert native[5:5] == str(big.sub(5, 5))
-    assert native[-5:-5] == str(big.sub(-5, -5))
-    assert native[2:-2] == str(big.sub(2, -2))
-    assert native[7:-7] == str(big.sub(7, -7))
-
-    assert native[5:3] == str(big.sub(5, 3))
-    assert native[5:7] == str(big.sub(5, 7))
-    assert native[5:-3] == str(big.sub(5, -3))
-    assert native[5:-7] == str(big.sub(5, -7))
-
-    assert native[-5:3] == str(big.sub(-5, 3))
-    assert native[-5:7] == str(big.sub(-5, 7))
-    assert native[-5:-3] == str(big.sub(-5, -3))
-    assert native[-5:-7] == str(big.sub(-5, -7))
-
-    assert native[2:] == str(big.sub(2))
-    assert native[:7] == str(big.sub(end=7))
-    assert native[-2:] == str(big.sub(-2))
-    assert native[:-7] == str(big.sub(end=-7))
-    assert native[:-10] == str(big.sub(end=-10))
-    assert native[:-1] == str(big.sub(end=-1))
+    assert native[0:5] == str(big.sub(0, 5)) and native[0:5] == str(big[0:5])
```

Comment:
Let's overload the `==` and `!=` operators, as well as the `__str__` dunder method. It will make this code much more concise and the `Str` class more usable.

Reply:
Added.